WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,232 @@
+    tick_thickness = 2,
+    tick_length = 30,
+  }
+end
+
+local properties = {
+  tick_visible = {
+    display = L["Visibility"],
+    setter = "SetVisible",
+    type = "bool",
+    defaultProperty = true,
+  },
+  tick_color = {
+    display = L["Color"],
+    setter = "SetTickColor",
+    type = "color",
+  },
+  tick_placement_mode = {
+    display = L["Placement Mode"],
+    setter = "SetTickPlacementMode",
+    type = "list",
+    values = WeakAuras.tick_placement_modes,
+  },
+  tick_placement = {
+    display = L["Placement"],
+    setter = "SetTickPlacement",
+    type = "number",
+    validate = WeakAuras.ValidateNumeric,
+  },
+  automatic_length = {
+    display = L["Automatic Length"],
+    setter = "SetAutomaticLength",
+    type = "bool",
+    defaultProperty = true,
+  },
+  tick_thickness = {
+    display = L["Thickness"],
+    setter = "SetTickThickness",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 2,
+  },
+  tick_length = {
+    display = L["Length"],
+    setter = "SetTickLength",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 30,
+  },
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+  end,
+  ["OrientationChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+    self:SetTickThickness(self.tick_thickness, true)
+  end,
+  ["OnSizeChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+  end,
+  SetVisible = function(self, visible)
+    if self.tick_visible ~= visible then
+      self.tick_visible = visible
+      if visible then
+        self:Show()
+      else
+        self:Hide()
+      end
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.tick_color[1], self.tick_color[2], self.tick_color[3], self.tick_color[4] = r, g, b, a or 1
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacementMode = function(self, placement_mode)
+    if self.placement_mode ~= placement_mode then
+      self.placement_mode = placement_mode
+      self:SetTickPlacement(self.tick_placement)
+    end
+  end,
+  SetTickPlacement = function(self, placement, forced)
```

Comment:
Though, I prefer if those funcitons had no forced parameter by splitting the functions into:

SetTickPlacement:
    if self.placement_mode ~= placement_mode then
      self.placement_mode = placement_mode
      self:UpdateTickPlacement()
    end

OrientationChanged()
  self.parentOrientation = self.parent.effectiveOrientation
  self.parentTrueWidth, self.parentTrueHeight = ...
  UpdateTickPlacement()

OnSizeChanged
  self.parentTrueWidth, self.parentTrueHeight = ...
  UpdateTIckPlacement()

UpdateTickPlacement()
    Basically using self's data to place the tick at the right place.